test/state-component-mocha.js

```diff
@@ -0,0 +1,139 @@
+                sendData(component, 'a', "A");
+                sendData(component, 'b', "B");
+            });
+        }).should.become({a: "A", b: "B"});
+    });
+    it("map indexed data together", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                a: {required: true, indexBy: 'aid'},
+                b: {required: true, indexBy: 'bid'}
+            },
+            onchange: function(state) {
+                handler(state);
+            }
+        }).then(stateComponent).then(createComponent).then(function(component){
+            // have the handler call a Promise resolve function to
+            // check that the data sent on the in port is passed to the handler
+            return new Promise(function(callback){
+                handler = callback;
+                sendData(component, 'a', {aid:"1", a: "A"});
+                sendData(component, 'a', {aid:"2", a: "A"});
+                sendData(component, 'a', {aid:"3", a: "A"});
+                sendData(component, 'b', {bid:"2", b: "B"});
+            });
+        }).should.become({a: {aid: "2", a: "A"}, b: {bid: "2", b: "B"}});
+    });
+    it("map indexed data by function", function() {
+        var handler;
+        return Promise.resolve({
+            inPorts:{
+                a: {required: true, indexBy: _.property('aid')},
+                b: {required: true, indexBy: _.property('bid')}
+            },
+            onchange: function(state) {
+                handler(state);
+            }
+        }).then(stateComponent).then(createComponent).then(function(component){
+            // have the handler call a Promise resolve function to
+            // check that the data sent on the in port is passed to the handler
+            return new Promise(function(callback){
+                handler = callback;
+                sendData(component, 'a', {aid:"1", a: "A"});
+                sendData(component, 'a', {aid:"2", a: "A"});
+                sendData(component, 'a', {aid:"3", a: "A"});
+                sendData(component, 'b', {bid:"2", b: "B"});
+            });
+        }).should.become({a: {aid: "2", a: "A"}, b: {bid: "2", b: "B"}});
+    });
+    function createComponent(getComponent) {
+        var component = getComponent();
+        _.forEach(component.inPorts, function(port, name) {
+            port.nodeInstance = component;
+            port.name = name;
+        });
+        _.forEach(component.outPorts, function(port, name) {
+            port.nodeInstance = component;
+            port.name = name;
+        });
+        return component;
+    }
+    function sendData(component, port, payload) {
+        var socket = noflo.internalSocket.createSocket();
+        component.inPorts[port].attach(socket);
+        socket.send(payload);
+        socket.disconnect();
+        component.inPorts[port].detach(socket);
+    }
+    function onceData(component, resolvePort, rejectPort, sendPort, sendPayload) {
```

Comment:
Is this function is actually used?   Did not look like it to me.  If not used, please remove.